layers.go

```diff
@@ -12,54 +12,67 @@ import (
 
 type bpLayersDir struct {
 	path   string
-	layers []bpLayer
+	layers map[string]bpLayer
 	name   string
 }
 
 func readBuildpackLayersDir(layersDir, buildpackID string) (bpLayersDir, error) {
 	path := filepath.Join(layersDir, buildpackID)
-	tomls, err := filepath.Glob(filepath.Join(path, "*.toml"))
-	if err != nil {
-		return bpLayersDir{}, err
-	}
-	layers := make([]bpLayer, 0, len(tomls))
-	for _, toml := range tomls {
-		name := strings.TrimRight(filepath.Base(toml), ".toml")
-		layers = append(layers, bpLayer{
-			layer{
-				path:       filepath.Join(layersDir, buildpackID, name),
-				identifier: fmt.Sprintf("%s/%s", buildpackID, name),
-			},
-		})
-	}
-	return bpLayersDir{
+	layers := map[string]bpLayer{}
+	bpDir := bpLayersDir{
 		name:   buildpackID,
 		path:   path,
 		layers: layers,
-	}, nil
+	}
+
+	fis, err := ioutil.ReadDir(path)
+	if err != nil && !os.IsNotExist(err) {
+		return bpDir, err
+	}
+	for _, fi := range fis {
+		if fi.IsDir() {
+			bpDir.layers[fi.Name()] = bpDir.newBPLayer(fi.Name())
+		}
+	}
+
+	tomls, err := filepath.Glob(filepath.Join(path, "*.toml"))
+	for _, toml := range tomls {
+		name := strings.TrimRight(filepath.Base(toml), ".toml")
+		if name == "launch" {
```

Comment:
It would probably make sense to remove this before it gets to the exporter and analyzer. (buildpack/lifecycle#70)

And/or explicitly test this as desired exporter behavior.

Reply:
+1

Reply:
It's interesting that `exporter` doesn't need the `launch.toml` files. That might imply that `builder` should clean them up